text/0000-copy-clone-semantics.md

```diff
@@ -0,0 +1,56 @@
+- Feature Name: N/A
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+With specialization on the way, we need to talk about the semantics of
+`<T as Clone>::clone() where T: Copy`.
+
+It's generally been an unspoken rule of Rust that a `clone` of a `Copy` type is
```

Comment:
You may want to cite https://github.com/rust-lang/rust/issues/23790, which initially proposed making `Clone` a supertrait of `Copy`, with the description:

> "Copy is a refinement of Clone where memcpy suffices"

The assumption that `Copy` and `Clone` would be compatible was also implicitly used in accepting [RFC 839](https://github.com/rust-lang/rfcs/blob/master/text/0839-embrace-extend-extinguish.md), as a premise thereof was the iea that it would be compatible to relax a `T: Copy` bound to a `T: Clone` bound later on.

In other words, it's been more than an unspoken rule that clone and copy must be compatible, even if the rule seems not to have made its way onto the documentation of the `Clone` trait itself.